### Assignment # 09 - Point Estimate and Interval Estimate (Confidence Interval)

A random survey of enrollment at **35** community colleges across the United States yielded the following figures: 

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 

1,263; 7,285; 28,165; 5,080; 11,622

Perform point estimate and interval estimate with **95% confidence level** using **t-distribution**.

Since we don't know the population variance, we use t-distribiution instead of normal distribution.

In [33]:
# Import Python packages

import pandas as pd 
import numpy as np 
from matplotlib import pyplot as pyplot 
import os, random 
import math
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline
import glob
import statistics as st
import scipy.stats as stats

### Step 0 - Data Preprocessing 

Process the raw data to make a list of integers. In order to calculate descriptive statistic, Python needs to work with a list of numbers.
### note:
Don't manually make the list by hand-typing the numbers. Write code to automate the data preparation.

In [34]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

data_1 = "6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;" 
data_2 = "5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; "
data_3 = "17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; "
data_4 = "1,263; 7,285; 28,165; 5,080; 11,622"
data = data_1 + data_2 + data_3 + data_4
data

'6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 1,263; 7,285; 28,165; 5,080; 11,622'

In [35]:
# Convert the single string to a list of strings using split() function
# Make sure to specify a delimter or separator
k = data.split(";")
print(k)


['6,414', ' 1,550', ' 2,109', ' 9,350', ' 21,828', ' 4,300', ' 5,944', ' 5,722', ' 2,825', ' 2,044', '5,481', ' 5,200', ' 5,853', ' 2,750', ' 10,012', ' 6,357', ' 27,000', ' 9,414', ' 7,681', ' 3,200', ' 17,500', ' 9,200', ' 7,380', ' 18,314', ' 6,557', ' 13,713', ' 17,768', ' 7,493', ' 2,771', ' 2,861', ' 1,263', ' 7,285', ' 28,165', ' 5,080', ' 11,622']


Create a list of integers from the list of strings using List Comprehension or for loop. Make sure to remove the "," first and then convert the strings to integers.

In [36]:
# Using List Comprehension
t = [int(i.replace(",", "")) for i in k]
print(t)

[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


In [37]:
# Use for loop

t = []
for i in k:
    y = i.replace(",", "")
    y = int(y)
    t.append(y)  
print(t)   

[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


### Step 1 - Calculate and Display the Sample Size and Sample Mean

In [38]:
# Calculate and display the sample size
sample_size = len(t)
print("Sample Size =", sample_size)


Sample Size = 35


In [39]:
# Calculate and display the sample mean
sample_mean = int(round(np.mean(t),0))
print("Sample Size =", sample_mean)


Sample Size = 8629


The point estimate of the mean enrollment of US community colleges is **8629**.

### Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error

Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default **Delta Degree of Freedom (DDOF)** for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify **ddof=1**. 

For the enrollment data, we round up the statistics to be the full integers (no decimal points).


In [51]:
# Calculate and display the sample standard deviation using Numpy's std function.
sample_std_dvn = int(round(np.std(t, ddof=1),0))
print("Sample Standard Deviation =",sample_std_dvn)


Sample Standard Deviation = 6944


In [50]:
# Calculate and display the sample standard error
sample_std_error = int(round(sample_std / math.sqrt(sample_size),0))      
print("Sample Standard Error =", sample_std_error)


Sample Standard Error = 1174


### Step 3 - Calculate t Critical Value using t-Distribution 

$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [52]:
# Calculate and display the t critical value using scipy.stats.t package ppf function
sigma = 1 -.95
significance_level = 1 - sigma/2
t_critical_value = round(stats.t.ppf(q=significance_level, df=sample_size-1),2)
print("t critical value =", t_critical_value)


t critical value = 2.03


### Step 4 - Calculate the Margin of Error

Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [53]:
# Calculate and display the margin of error
margin_of_error = int(round(t_critical_value * std_err,2))
print("Margin of Error =", margin_of_error)


Margin of Error = 2383


### Step 5 - Calculate Lower and Upper Limit of the Confidence Interval

Lower Limit = Sample Mean - Margin of Error

Upper Limit = Sample Mean + Margin of Error

In [54]:
# Calculate and display the lower limit
lower_limit = sample_mean - margin_of_error
print ("Lower Limit = ", lower_limit)


Lower Limit =  6246


In [55]:
# Calculate and display the upper limit
upper_limit = sample_mean + margin_of_error
print ("Upper Limit = ", upper_limit)


Upper Limit =  11012


### Step 6 - Now We have the 95% Confidence Interval
Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [57]:
# Print the 95% confidence interveral in the form (lower limit, upper limit)
print("The 95% Confidence Interval = (", lower_limit, ",", upper_limit, ")")


The 95% Confidence Interval = ( 6246 , 11012 )


### The End